File: challenge.js
/**
 * @Author: xiaojiezhang
 * @Date:   2019-01-14T09:01:25-05:00
 * @Last modified by:   xiaojiezhang
 * @Last modified time: 2019-01-14T09:30:02-05:00
 */



'use strict'

const any = function (arr, fun) {
  return arr.some(fun)
}

const none = function (arr, fun) {
  return !arr.some(fun)
}

// Bonuses

const one = function (arr, fun) {
  return arr.filter(fun).length === 1 ? true : false
}

const all = function (arr, fun) {
  return arr.every(fun)
}

// Do not change this.
module.exports = {
  any,
  none,
  one,
  all
}

File: challenge.rb
# @Author: xiaojiezhang
# @Date:   2019-01-14T09:01:25-05:00
# @Last modified by:   xiaojiezhang
# @Last modified time: 2019-01-14T09:40:57-05:00



# frozen_string_literal: true

# Add your function parameters between the parentheses.
def any? (list,&block)
  list.any?(&block)
end

# Add your function parameters between the parentheses.
def none?(list, &block)
  ! list.any?(&block)
end

# Bonuses

# Add your function parameters between the parentheses.
def one?(list, &block)
  list.one?(&block)
end

# Add your function parameters between the parentheses.
def all?(list, &block)
  array.all?(&block)
end
